plugins/emmc/fu-emmc-device.c

```diff
@@ -217,6 +217,15 @@ fu_emmc_device_probe(FuDevice *device, GError **error)
 		return FALSE;
 	if (!fu_emmc_device_get_sysattr_guint64(udev_parent, "oemid", &oemid, error))
 		return FALSE;
+
+	/*
+	 * TODO: Remove this once the Linux kernel bug is fixed:
+	 * The mmc driver wrongly parse the CID register picking 16 bits for the oemid instead of
+	 * just 8 - see drivers/mmc/core/mmc.c:107. For further reference see the eMMC spec
+	 * JESD84-B51 paragraph 7.2.
+	 */
+	oemid &= 0xff;
```

Comment:
This will probably break the GUID matching, this is how GUIDs look like for one of WDC devices

cd0cd165-b9b4-521b-bebf-49a7d263a2c5 <- EMMC\NAME_DA4032
5a02344d-585b-507b-ad59-19e96c03ca46 <- EMMC\MAN_0045&OEM_0100
d5927774-7e99-5ffe-ad3d-5946a78ce9d1 <- EMMC\MAN_0045&OEM_0100&NAME_DA4032
d50a8bba-268b-5b39-8b5b-6c6d300667e1 <- EMMC\MAN_0045&NAME_DA4032&REV_0x3536323330613137
1eb265a9-8f5d-5c58-b599-2eacd370d0f4 <- EMMC\MAN_0045&OEM_0100&NAME_DA4032&REV_0x3536323330613137
5500994c-1b73-52b4-acf4-bb574b6f3029 <- EMMC\0069&0256&DA4032

Reply:
And this exactly what I am trying to fix.
Our metainfo file will designate the correct oemid e.g. EMMC\MAN_0045&OEM_0000 with the correct GUID...

Reply:
I think what @daniil-lunev is saying is that existing firmware is using that GUID for matching -- I think in this case it's a good idea to just create an extra instance ID rather than bitmask the existing one. A FuDevice can have as many Instance IDs (that get converted to GUIDs) as it likes.

Reply:
got it.  Thanks.  Will re-work it in my next spin.

Reply:
On a second thought - I will just drop this.
I already fixed this in the kernel, and practically have no use for it now.

Reply:
Oh, if you changed it in the kernel - that likely means the matching will be broken anyway.
We would need to re-mirror the cabs on the ChromeOS side once they are updated to mention new GUIDs.
If we are updating cabs - I would also want to address the issue with FW version ids - the way they are now they are not consistently matching "new version - higher number", and thus we needed to do something ugly for the updates to work. We would want to have some kind of quirk based mapping for ids like "0x3536323330613137" to corresponding versions (1.1 for this particular ID if I am not mistaken). I will get to you about it once I am back from my vacation.

Reply:
Thanks.  The quirk doesn't concerns our DA4* devices but only the 4 devices listed in the quirk file: DI4* & DJ4*.
I don't think we even have cabs for those yet.

Reply:
> I already fixed this in the kernel, and practically have no use for it now.

If you haven't already; I suggest that you get the kernel fix backported to all the applicable stable trees.  As many distros track the stable trees this will make it less likely to encounter the problematic value.

Reply:
Yeah - I cc'ed stable.  see e.g. https://www.spinics.net/lists/stable/msg686502.html
Thanks,
Avri

Reply:
Daniil, just to make you aware of https://www.spinics.net/lists/linux-mmc/msg78222.html should you missed it.
The oemid fixed was already got picked up to various stable branches, up to including 4.14.

Thanks,
Avri

Reply:
but got reverted just now...

Thanks,
Avri